### PR TITLE
user-profile fetch updated

### PR DESCRIPTION
**Pull request by Prajwal17Tunerlabs**
# Description
These recommendations are intended to promote code quality and team communication during software development. They cover a variety of topics, including ensuring that pull requests are submitted to the correct branch, documenting new methods, preserving consistency across many services, and avoiding typical blunders like accessing APIs or DB queries within loops. Sensitive data should not be uploaded, and changes to environment variables or database models should be executed consistently. Teams may work more effectively and develop higher-quality software by adhering to these standards.

## Type of change
Please choose appropriate options.

- [ ]  Bug fix (non-breaking change which fixes an issue)
- [ ]  New feature (non-breaking change which adds functionality)
- [ ]  Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ]  Enhancement (additive changes to improve performance)
- [ ]  This change requires a documentation update

## Checklist 

- [ ]  It's critical to avoid making needless file modifications in contributions, such as adding new lines, console logs, or additional spaces, to guarantee cleaner and more efficient code. Furthermore, eliminating unnecessary imports from a file might enhance code readability and efficiency.
- [ ]  Ensure that the pull request is assigned to the right base branch and that the development branch name contains the JIRA Task Id. Furthermore, each commit message should include the JIRA Task Id in the manner "ED-100: message".
- [ ]  Only update packages if it is mentioned and authorized in the design document, and make sure that you have the required permissions.
- [ ]  Avoid making API and database queries inside a loop as it can lead to performance issues and slow down the system.
- [ ]  When calling another function inside a given function, add comments explaining the purpose and meaning of the passed arguments and expected return values.
- [ ]  If adding a blank argument in a function, add a comment explaining the reason for the blank argument.
- [ ]  Before submitting a pull request, do a self-review of your code to ensure there are no conflicts with the base branch and all comments have been addressed.
- [ ]  Before merging a pull request, it's important to have other team members review it to catch any potential errors or issues
- [ ]  To maintain code integrity, it's important to remove all related changes when removing code during a code review.
- [ ]  If new constants, endpoints, or utility functions are introduced, it is important to check if they already exist in the service to avoid any duplication.
- [ ]  Whenever a new environment variable is added to a service, it's important to ensure that the necessary changes are made to related files such as ".env.sample" and "envVariables.js" to maintain consistency and avoid errors. Additionally, the new environment variable should be added to the devops repository to ensure that it is properly documented and accessible to the team.
- [ ]  When adding a new function to a service, it is important to document it with relevant information such as the name, parameters, and return value in a consistent format across all services. Additionally, if there are any changes to the API response, ensure that the documentation in the controllers is updated accordingly. 
- [ ]  Write a clear and concise commit message that describes the changes made.
- [ ]  Maintain consistent function signature and code across all services when adding a function to multiple services. Implement changes to database models in all services that use the same model. 
- [ ]  Use only let and const. Do not use var.
- [ ]  Make common functions for repetitive code blocks.
- [ ]  Avoid uploading sensitive information such as secret tokens or passwords in pull requests to ensure data security.
- [ ]  Maintain consistent indentation and spacing throughout the code.

### CODE REVIEW

File: generics/services/projects.js [hunk 1]
```diff
@@ -0,0 +1,62 @@
+/**
+ * name : projects.js
+ * author : prajwal
+ * Date : 09-September-2024
+ * Description : All projects related api call.
+ */
+
+const request = require('request')
+const projectServiceUrl = process.env.PROJECT_SERVICE_URL
```
Comment (VISHNUDAS-tunerlabs) [inline generics/services/projects.js:9]: @Prajwal17Tunerlabs Please make changes to the envs as per we discussed

File: controllers/v1/userProjects.js [hunk 2]
```diff
@@ -1199,7 +1199,10 @@ module.exports = class UserProjects extends Abstract {
 		return new Promise(async (resolve, reject) => {
 			try {
 				// ReIssue certificate of given project : projectId is passed as param
-				let projectDetails = await userProjectsHelper.certificateReIssue(req.params._id)
+				let projectDetails = await userProjectsHelper.certificateReIssue(
```
Comment (VISHNUDAS-tunerlabs) [inline controllers/v1/userProjects.js:1202]: @Prajwal17Tunerlabs if we are not storing the user Profile , it s better to call user service directly

**Review by VISHNUDAS-tunerlabs — CHANGES_REQUESTED** [review]
Reviewed Sep 10

File: controllers/v1/userProjects.js (hunk 2 above)
Comment (Prajwal17Tunerlabs) [reply controllers/v1/userProjects.js:1202]: user service changes for certificateReissue api reverted.

File: .env.sample [hunk 3]
```diff
@@ -72,7 +72,11 @@ DOWNLOADABLE_URL_EXPIRY_IN_SECONDS = 300
...
-TIMEZONE_DIFFRENECE_BETWEEN_LOCAL_TIME_AND_UTC= +05:30 
+TIMEZONE_DIFFRENECE_BETWEEN_LOCAL_TIME_AND_UTC = +05:30 
 
 #API doc
-API_DOC_URL= "/project/api-doc"
\ No newline at end of file
+API_DOC_URL= "/project/api-doc"
+
+INTERFACE_SERVICE_URL = "http://localhost:9000"                                  // Interface service base url
+USER_SERVICE_NAME = "/user"                                                      // User service name
```
Comment (VISHNUDAS-tunerlabs) [inline .env.sample:81]: @Prajwal17Tunerlabs can we make thisUSER_SERVICE_BASE_URL please

File: .env.sample [hunk 4]
```diff
@@ -72,7 +72,11 @@ DOWNLOADABLE_URL_EXPIRY_IN_SECONDS = 300
...
+ENTITY_MANAGEMENT_SERVICE_NAME = "/entity-management"                            // Entity management service name
```
Comment (VISHNUDAS-tunerlabs) [inline .env.sample:82]: @Prajwal17Tunerlabs ENTITY_MANAGEMENT_SERVICE_BASE_URL

File: envVariables.js [hunk 5]
```diff
@@ -123,6 +123,18 @@ let enviromentVariables = {
 		message: 'Required api doc url',
 		optional: false,
 	},
+	INTERFACE_SERVICE_URL: {
+		message: 'Interface service url required',
+		optional: false,
+	},
+	USER_SERVICE_NAME: {
```
Comment (VISHNUDAS-tunerlabs) [inline envVariables.js:130]: @Prajwal17Tunerlabs please update the variable names here as well

File: generics/services/entity-management.js [hunk 6]
```diff
@@ -25,7 +25,10 @@ const entityDocuments = function (filterData = 'all', projection = 'all') {
 	return new Promise(async (resolve, reject) => {
 		try {
 			// Function to find entity documents based on the given filter and projection
-			const url = entityManagementServiceUrl + CONSTANTS.endpoints.FIND_ENTITY_DOCUMENTS
+			const url =
+				interfaceServiceUrl +
+				process.env.ENTITY_MANAGEMENT_SERVICE_NAME +
```
Comment (VISHNUDAS-tunerlabs) [inline generics/services/entity-management.js:30]: @Prajwal17Tunerlabs please update the variable names and test

**Review by VISHNUDAS-tunerlabs — CHANGES_REQUESTED** [review]
Reviewed on 11-sept 11:15 AM

File: .env.sample (hunk 3 above)
Comment (Prajwal17Tunerlabs) [reply .env.sample:81]: I have changed the env variable name to the proposed one.

File: .env.sample (hunk 4 above)
Comment (Prajwal17Tunerlabs) [reply .env.sample:82]: I have changed the env variable name to the proposed one.

File: envVariables.js (hunk 5 above)
Comment (Prajwal17Tunerlabs) [reply envVariables.js:130]: env variable names are updated in envVariable.js file.

File: generics/services/entity-management.js (hunk 6 above)
Comment (Prajwal17Tunerlabs) [reply generics/services/entity-management.js:30]: env variable names are updated .

**Pull request merged**